bin/inx-chronicle/src/check_health.rs

```diff
@@ -0,0 +1,35 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use async_trait::async_trait;
+use chronicle::runtime::{Actor, HandleEvent, RuntimeError, ScopeView, Sender};
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub(crate) enum CheckHealthError {
+    #[error(transparent)]
+    Recv(#[from] tokio::sync::oneshot::error::RecvError),
+    #[error(transparent)]
+    Runtime(#[from] RuntimeError),
+}
+
+pub(crate) struct IsHealthy(pub tokio::sync::oneshot::Sender<bool>);
+
+#[async_trait]
+pub(crate) trait CheckHealth {
+    async fn is_healthy<A: Actor>(&self) -> Result<bool, CheckHealthError>
+    where
+        A: 'static + HandleEvent<IsHealthy>;
+}
+
+#[async_trait]
+impl CheckHealth for ScopeView {
+    async fn is_healthy<A: Actor>(&self) -> Result<bool, CheckHealthError>
+    where
+        A: 'static + HandleEvent<IsHealthy>,
+    {
+        let (sender, receiver) = tokio::sync::oneshot::channel();
+        self.addr::<A>().await.send(IsHealthy(sender))?;
+        Ok(receiver.await?)
```

Comment:
You something possible hang due to Chronicle being unhealthy making this await point block forever? Or in other words: Should we race this await point against a health-check timeout which yields false if the health check wasn't performed in time?

Reply:
I've thought about it, because currently hornet does block forever due to the deadlock issue, but tbh that's not something we should have to account for. Hornet needs to fix their mess.